### PR TITLE
Fix configs to properly use pytorch-lightning==1.6 with GPU

### LINKED ISSUE #232: GPU available but not used
Executing CFLOW model on Colab, PyTorch returns this

```
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs

GPU available but not used. Set "accelerator" and "devices" using Trainer(accelerator='gpu', devices=1).
```

Even if i'm setting gpus=1 on the config.yml of cflow.

 
**To Reproduce**
`train.py --model cflow
`

**GPU Model**
GPU 0: Tesla T4

**Versions**
```
pytorch-lightning             1.6.1
torch                         1.10.0+cu111
torchaudio                    0.10.0+cu111
torchmetrics                  0.7.3
torchsummary                  1.5.1
torchtext                     0.11.0
torchvision                   0.11.1+cu111
```

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- Fix configs to properly use `pytorch-lightning==1.6.*` and train with GPU

- Fixes #232 

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/models/cflow/config.yaml [hunk 1]
```diff
@@ -55,10 +55,12 @@ trainer:
   benchmark: false
   check_val_every_n_epoch: 1
   default_root_dir: null
+  detect_anomaly: false
   deterministic: false
   enable_checkpointing: true
+  enable_progress_bar: true
   fast_dev_run: false
-  gpus: 1
+  gpus: null # Set automatically
```
Comment (ashwinvaidya17) [inline anomalib/models/cflow/config.yaml:63]: Does it mean that pytorch lightning will distribute job across all available GPUs? In which case, we might have to look at the performance.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks! One minor comment but rest looks good

File: anomalib/models/cflow/config.yaml (hunk 1 above)
Comment (samet-akcay) [reply anomalib/models/cflow/config.yaml:63]: Not sure what's the best approach here. Setting it to 1 or automatically assigning it?
@ashwinvaidya17, @djdameln?

Comment (ashwinvaidya17) [reply anomalib/models/cflow/config.yaml:64]: I would prefer using only a single gpu by default. I feel users should use distribute only if they know what they are doing. If we distribute the training then we will have to look at the learning rate and experiments might not be reproducible across different number of GPUs.

Comment (djdameln) [reply anomalib/models/cflow/config.yaml:64]: What if no GPUs are available. Will it break or just switch to CPU training?

Comment (samet-akcay) [reply anomalib/models/cflow/config.yaml:64]: > What if no GPUs are available. Will it break or just switch to CPU training?

It switches to CPU.

> I would prefer using only a single gpu by default. I feel users should use distribute only if they know what they are doing. If we distribute the training then we will have to look at the learning rate and experiments might not be reproducible across different number of GPUs.

As far as I understand, this doesn't set it to distributed training because `strategy` is still `null`. It automatically uses a single GPU.

Comment (ashwinvaidya17) [reply anomalib/models/cflow/config.yaml:64]: In that case I am fine with how it is now.

**Comment by ashwinvaidya17** [comment]
@aj2563 This is a known issue and is addressed in a different open PR. For now you can use `pip install torchmetrics==0.6.0` to make it work. Here is a colab link which uses this branch for training. https://colab.research.google.com/drive/18NciqtQwlrUIiuwWn8ld_Tp_5K6kgRPH?usp=sharing The first 7 cells are relevant in your case.

**Review by djdameln — APPROVED** [review]

**Comment by ING3TP2** [comment]
Hello, I am having the error: Cannot import name 'configure_logger'  from 'anomalib.utils.loggers' while trying to run the following command : 
`python tools/train.py --config anomalib/models/padim/config.yaml`

image
